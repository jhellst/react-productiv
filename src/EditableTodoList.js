import React from "react";
import EditableTodo from "./EditableTodo";
import { v4 as uuid } from 'uuid';

/** Show list of editable todos.
 *
 * Props:
 * - todos: array of [ todo, ... ]
 * - update(): fn to call to update a todo
 * - remove(): fn to call to remove a todo
 *
 * TodoApp -> EditableTodoList -> [ EditableTodo, ... ]
 */

function EditableTodoList({ todos, update, remove }) {
  return (
    <div>
      {/*   FIXME  */}
      {todos.map(todo =>
        <EditableTodo
          key = {uuid()}
          todo={todo}
          updateTodo={update}
          deleteTodo={remove}
        />)}
    </div>
  );
}

export default EditableTodoList;
