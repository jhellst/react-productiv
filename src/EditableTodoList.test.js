import React from "react";
import { render } from "@testing-library/react";
import App from "./App";
import EditableTodoList from "./EditableTodoList";

describe("EditableTodoList Component", function () {
  it("renders without crashing", function () {
    render(<EditableTodoList todos={[{
      id: "1",
      title: "test",
      description: "testDescription"
    }]} />);
  });

  it("matches snapshot", function () {
    const { container } = render(<EditableTodoList todos={[{
      id: "1",
      title: "test",
      description: "testDescription"
    }]} />);
    expect(container).toMatchSnapshot();
  });

  it("Todo contains provided information", function () {
    const { container } = render(<EditableTodoList todos={[{
      id: "1",
      title: "test",
      description: "testDescription"
    }]} />);

    expect(container).toContainHTML("test");
    expect(container).toContainHTML("testDescription");
  });


});