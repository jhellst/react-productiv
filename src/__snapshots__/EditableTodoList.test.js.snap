// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTodoList Component matches snapshot 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="1"
        >
          <div>
            <b>
              test
            </b>
             
            <small>
              (priority: 
              )
            </small>
          </div>
          <div>
            <small>
              testDescription
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
